php/class-customize-snapshot-manager-back-compat.php

```diff
@@ -0,0 +1,1009 @@
+			return;
+		}
+
+		$this->add_widget_setting_preview_filters();
+		$this->add_nav_menu_setting_preview_filters();
+
+		/*
+		 * Populate post values.
+		 *
+		 * Note we have to defer until setup_theme since the transaction
+		 * can be set beforehand, and wp_magic_quotes() would not have
+		 * been called yet, resulting in a $_POST['customized'] that is
+		 * double-escaped. Note that this happens at priority 1, which
+		 * is immediately after Customize_Snapshot_Manager::store_customized_post_data
+		 * which happens at setup_theme priority 0, so that the initial
+		 * POST data can be preserved.
+		 */
+		if ( did_action( 'setup_theme' ) ) {
+			$this->import_snapshot_data();
+		} else {
+			add_action( 'setup_theme', array( $this, 'import_snapshot_data' ) );
+		}
+
+		// Block the robots.
+		add_action( 'wp_head', 'wp_no_robots' );
+
+		// Preview post values.
+		if ( did_action( 'wp_loaded' ) ) {
+			$this->preview_snapshot_settings();
+		} else {
+			add_action( 'wp_loaded', array( $this, 'preview_snapshot_settings' ), 11 );
+		}
+	}
+
+	/**
+	 * Populate post values and $_POST['customized'] wth the snapshot's data.
+	 *
+	 * Plugins used to have to dynamically register settings by inspecting the
+	 * $_POST['customized'] var and manually re-parse and inspect to see if it
+	 * contains settings that wouldn't be registered otherwise. This ensures
+	 * that these plugins will continue to work.
+	 *
+	 * Note that this can't be called prior to the setup_theme action or else
+	 * magic quotes may end up getting added twice.
+	 *
+	 * @see Customize_Snapshot_Manager::should_import_and_preview_snapshot()
+	 */
+	public function import_snapshot_data() {
+		/*
+		 * We don't merge the snapshot data with any existing existing unsanitized
+		 * post values since should_import_and_preview_snapshot returns false if
+		 * there is any existing data in the Customizer state. This is to prevent
+		 * clobbering existing values (or previewing non-snapshotted values on frontend).
+		 * Note that wp.customize.Snapshots.extendPreviewerQuery() will extend the
+		 * previewer data to include the current snapshot UUID.
+		 */
+		$snapshot_values = array_filter(
+			wp_list_pluck( $this->snapshot->data(), 'value' ),
+			function( $value ) {
+				return ! is_null( $value );
+			}
+		);
+
+		// Populate input vars for back-compat.
+		$_POST['customized'] = wp_slash( wp_json_encode( $snapshot_values ) );
+		// @codingStandardsIgnoreStart
+		$_REQUEST['customized'] = $_POST['customized'];
+		// @codingStandardsIgnoreEnd
+
+		foreach ( $snapshot_values as $setting_id => $value ) {
+			$this->customize_manager->set_post_value( $setting_id, $value );
+		}
+	}
+
+	/**
+	 * Determine whether the current snapshot can be previewed.
+	 *
+	 * @param Customize_Snapshot $snapshot Snapshot to check.
+	 * @return true|\WP_Error Returns true if previewable, or `WP_Error` if cannot.
+	 */
+	public function should_import_and_preview_snapshot( Customize_Snapshot $snapshot ) {
```

Comment:
This param should probably be explicitly a `Customize_Snapshot_Back_Compat` since the `saved` method is only defined on it.